File: wordcount.hs
#!/usr/bin/env runhaskell

{-|

Module      : WordCount
Description : Count words from file read from STDIN.
Copyright   : © Frank Jung, 2019
License     : GPL-3

Word Count as per <https://linux.die.net/man/1/wc wc(1)>.

>>> wc -w wordcount.hs
38 wordcount.hs

>>> cat wordcount.hs | runhaskell wordcount.hs
38

-}

module WordCount (main) where

-- | Count words from file read from STDIN.
main :: IO ()
main = interact $ show . length . words

File: wordcount.py
#!/usr/bin/env python3

"""
Word count as per wc(1).

Count words from file read from stdin.

To run:

    $ cat wordcount.hs | python3 ./wordcount.py
    38
"""

import sys
from functools import reduce


def word_count(stream: 'file') -> 'int':
    """
    Count words from STDIN IO stream.
    """
    return reduce(lambda x, y: x + y, list(map(lambda x: len(x.split()),
                                               stream)))


if __name__ == '__main__':
    print(word_count(sys.stdin.readlines()))

